tests/system.rs

```diff
@@ -30,8 +30,23 @@ use rustfmt::rustfmt_diff::*;
 
 const DIFF_CONTEXT_SIZE: usize = 3;
 
-fn get_path_string(dir_entry: io::Result<fs::DirEntry>) -> PathBuf {
-    dir_entry.expect("Couldn't get DirEntry").path().to_owned()
+// Returns a `Vec` containing `PathBuf`s of files with a rs extension in the
+// given path. The `recursive` argument controls if files from subdirectories
+// are also returned.
+fn get_test_files(path: &Path, recursive: bool) -> Vec<PathBuf> {
+    let mut files = vec![];
+    if path.is_dir() {
+        for entry in fs::read_dir(path).expect("Couldn't read source dir") {
```

Comment:
Bikeshedding, the given path is not necessarily a source directory.

Reply:
True! How do you feel about the following?
```diff
@@ -36,7 +36,10 @@ const DIFF_CONTEXT_SIZE: usize = 3;
 fn get_test_files(path: &Path, recursive: bool) -> Vec<PathBuf> {
     let mut files = vec![];
     if path.is_dir() {
-        for entry in fs::read_dir(path).expect("Couldn't read source dir") {
+        for entry in fs::read_dir(path).expect(&format!(
+            "Couldn't read directory {}",
+            path.to_str().unwrap()
+        )) {
             let entry = entry.expect("Couldn't get DirEntry");
             let path = entry.path();
             if path.is_dir() && recursive {
```

Reply:
I am sorry for the late reply. That looks good to me!

Reply:
No worries about the delay. I appreciate your feedback!